crates/ty_server/src/server/api/diagnostics.rs

```diff
@@ -26,17 +26,23 @@ pub(super) struct Diagnostics<'a> {
 }
 
 impl Diagnostics<'_> {
-    pub(super) fn result_id_from_hash(diagnostics: &[ruff_db::diagnostic::Diagnostic]) -> String {
+    pub(super) fn result_id_from_hash(
```

Comment:
nit: It might be useful to document when the return type will be `None` and `Some`